### ISSUE TITLE
Inconsistent project name when updating

### ISSUE THREAD

**Issue by FlorianWilhelm** [pyscaffold/pyscaffold#219]
First of all, this is only an edge case problem. During an update right now we make the basic assumption that the name of the directory holding the project is still the project name. In most cases this will be true but not every time. The problem is that we treat the positional argument during an update as project name but in reality it is only the directory holding the project. The actual project name is saved within setup.cfg. To solve this edge case we need to differentiate between those two concepts. This is closely related to issue #138 which can only be cleanly resolved if we differentiate between project directory and project name.

My suggestion for version 4.0 would be following API change. The positional PROJECT argument of putup will be renamed into PATH thus specifying the directory for the project. An additional optional `--project` argument specifies the actual project name. If the project name is not provided the last path element of PATH is used. Thus for most users the behaviour will not change but we have a clear separation between project directory and project name. 

## Versions and main components

* PyScaffold Version: all
* Python Version: all
* Operating system: all

**Comment by FlorianWilhelm** [comment]
@abravalheri What do you think about that? If you agree I would also postpone #138 to version 4.0 since with the current internals it cannot be implemented without nasty hacks.

**Comment by abravalheri** [comment]
@FlorianWilhelm, let me just try to understand something first.
Given the scenario you have just described, `project` for us will be the name of the thing we install with `pip`, right?

For example, right now if we do:
```sh
$ putup pyscaffoldext-myextension --namespace pyscaffold --package myextension
```
`pyscaffoldext-myextension` is both `project` and directory name. Therefore we install it using
```sh
$ pip install pyscaffoldext-myextension
```

The proposal is that, we could do:
```sh
$ putup myfolder --project pyscaffoldext-myextension --namespace pyscaffold --package myextension
...
$ pip install pyscaffoldext-myextension
```
Is that right?

**Comment by abravalheri** [comment]
It is ok for me, and indeed it brings more clarity when thinking about paths and specially `putup -u .`.
However, since we are talking about naming concepts, I feel a little bit weird using `project` as the main name of the thing that you upload to the Python Package Index (one can argue that you should you should submit `packages` to PyPI).

Since we don't have used `-n` or `-i` yet, we could think about `--name` or `--id`. What do you think? (I agree in changing PROJECT to PATH).

(Naming is definitely the hardest problem in computing).

**Comment by FlorianWilhelm** [comment]
@abravalheri Yes, exactly like you described it. I agree that `--name` is even better since this is also the attribute name in `setup.cfg` which is then more consistent. I think most people will have path = project name = package name anyway. Scikit-learn is a famous example where you `pip install scikit-learn` but import `sklearn`, thus project-name != package name. Maybe we have to check our current documentation if this is really clear.

**Comment by abravalheri** [comment]
I agree Florian.

Something to consider is that, if we are accepting any path, for example:
```bash
$ putup /media/sf_mounted_folder/myproj
```
it makes more sense to start the scaffold dictionary directly under the project folder, instead of the parent folder as we do today.

Therefore, instead of:
```python
struct = {
    opts['project']: {
        'README.rst': (templates.readme(opts), FileOp.NO_OVERWRITE),
        '.gitignore': (templates.gitignore(opts), FileOp.NO_OVERWRITE),
        'src': { 
            ...
```
it would be more meaningful (and simplify things a bit), if we had
```python
struct = {
    'README.rst': (templates.readme(opts), FileOp.NO_OVERWRITE),
    '.gitignore': (templates.gitignore(opts), FileOp.NO_OVERWRITE),
    'src': { 
         ...
```

**Comment by FlorianWilhelm** [comment]
Yes, I would like to see that. It becomes simpler, more consistent and thus more beautiful. I think we have a plan now :-)

**Comment by abravalheri** [comment]
Addressed by #241
